ansible/playbooks/do_setup.yml

```diff
@@ -34,18 +34,22 @@
       ansible.builtin.command: doctl databases create {{ db_name }} --region {{ droplet_region }} --engine mongodb --version 6 --output json
       async: 1800
       poll: 60
-      register: db_create_result
-      when: "db_check.stdout.find(' ' + db_name + ' ') == -1"
+      register: db_create
+      when: "db_check.stdout | from_json | json_query(name_query) | length < 1"
+      vars:
+        name_query: '[?name==`{{db_name}}`]'
 
     - name: d_ocean | db | set db url
       ansible.builtin.set_fact:
-        db_url: "{{ db_create_result.stdout | from_json | json_query('[0].private_connection.uri') }}"
-      when: "db_check.stdout.find(' ' + db_name + ' ') == -1"
+        db_url:  "{{ db_create.stdout if db_create.changed == false else db_check.stdout | from_json | json_query(name_query) | json_query('[0].private_connection.uri') }}" 
```

Comment:
just wondering what this does if db_create.changed is false.. it assigns the db_url to `db_create.stdout` or just skips this?

Reply:
If it's false it uses the `db_check` register which contains the output of `doctl db list -o json`, which has the same return data as the creation API, so the rest of the pipeline can be the same.